phone_actions.rb

```diff
@@ -0,0 +1,15 @@
+class PhoneActions
+
+  def initialize(username, password)
+    @actions_url = "https://#{username}:#{password}@api.<made_up>.com/actions/"
+  end
+
+  def call!(number, message)
+    JSON.parse(RestClient.post("@actions_url/#{text}", {message: message, number: number}))
+  end
+
+  def text!(number, message)
+    JSON.parse(RestClient.post("@actions_url/#{call}", {message: message, number: number}))
```

Comment:
It has been a bit since I used `RestClient`, but I think `post` and its other methods return a response object, not the actual content. You'd need to get the response, and then parse the body.

```suggestion
    response = RestClient.post("@actions_url/#{call}", {message: message, number: number})
    JSON.parse(response.body)
```

Do we know what the API is supposed to return? I am wondering if it returns a different status code if the message or number of not formed a specific way.

It's also possible that it returns invalid JSON, like if you get a generic html page, and that would fail.

Given that, it may be worth making a helper method like `post` that handles checking the http status, and that it parses the response. It'd be a good place to raise our own error class that other parts of the code base can then rescue.